src/groupeddataframe/splitapplycombine.jl

```diff
@@ -236,6 +237,74 @@ function _combine_process_agg((cs_i,)::Ref{Any},
     end
 end
 
+function _combine_process_groupindices((cs_i,)::Ref{Any},
+                                       optional_i::Bool,
+                                       parentdf::AbstractDataFrame,
+                                       gd::GroupedDataFrame,
+                                       seen_cols::Dict{Symbol, Tuple{Bool, Int}},
+                                       trans_res::Vector{TransformationResult},
+                                       idx_agg::Vector{Int})
+    @assert cs_i isa Pair{Vector{Int}, Pair{typeof(groupindices), Symbol}}
+    @assert first(cs_i) == Int[]
+    @assert !optional_i
+    @assert idx_agg !== NOTHING_IDX_AGG
+    out_col_name = last(last(cs_i))
+    outcol = 1:length(gd)
+
+    return function()
+        if haskey(seen_cols, out_col_name)
+            optional, loc = seen_cols[out_col_name]
+            # we have seen this col but it is not allowed to replace it
+            optional || throw(ArgumentError("duplicate output column name: :$out_col_name"))
+            @assert trans_res[loc].optional && trans_res[loc].name == out_col_name
+            trans_res[loc] = TransformationResult(idx_agg, outcol, out_col_name, optional_i)
+            seen_cols[out_col_name] = (optional_i, loc)
+        else
+            push!(trans_res, TransformationResult(idx_agg, outcol, out_col_name, optional_i))
+            seen_cols[out_col_name] = (optional_i, length(trans_res))
+        end
+    end
+end
+
+function _combine_process_proprow((cs_i,)::Ref{Any},
+                                  optional_i::Bool,
+                                  parentdf::AbstractDataFrame,
+                                  gd::GroupedDataFrame,
+                                  seen_cols::Dict{Symbol, Tuple{Bool, Int}},
+                                  trans_res::Vector{TransformationResult},
+                                  idx_agg::Vector{Int})
+    @assert cs_i isa Pair{Vector{Int}, Pair{typeof(proprow), Symbol}}
+    @assert first(cs_i) == Int[]
+    @assert !optional_i
+    @assert idx_agg !== NOTHING_IDX_AGG
+    out_col_name = last(last(cs_i))
+
+    if getfield(gd, :idx) === nothing
+        lens = zeros(Int, length(gd))
+        @inbounds for gix in gd.groups
+            gix > 0 && (lens[gix] += 1)
+        end
+        outcol = lens / sum(lens)
+    else
+        lens2 = gd.ends .- gd.starts .+ 1
+        outcol = lens2 / sum(lens2)
```

Comment:
Likewise, could be interesting to test:
```suggestion
        outcol = similar(gd.ends, Float64)
        outcol .= gd.ends .- gd.starts .+ 1
        outcol ./= sum(outcol)
```

Reply:
This is faster - I will apply it.